### PR TITLE
network-manager: update to 1.0.0

### PR DESCRIPTION
**Pull request by cstrahan**
This is a work-in-progress to update NetworkManger to 1.0.0.

### CODE REVIEW

**Comment by iElectric** [comment]
nice! did you test nm-applet with virtualboxHost enabled? it might display all vbox interfaces

File: pkgs/tools/networking/network-manager-applet/default.nix [hunk 1]
```diff
@@ -27,29 +30,44 @@ stdenv.mkDerivation rec {
...
+    rm -r $out/libexec
+    rm -r $out/share
+    rm -r $out/etc
+    rm -r $out/lib/gdk-pixbuf-loaders-2.0
   '';
 
-  preFixup = ''
+  postFixup = lib.optionalString applet ''
+    for prog in $out/bin/nm-applet $out/bin/nm-connection-editor $out/libexec/nm-applet-migration-tool; do
+      patchelf --set-rpath "$(patchelf --print-rpath $prog):${libnm-gtk}/lib" $prog
```
Comment (lethalman) [inline pkgs/tools/networking/network-manager-applet/default.nix]: Serious? Something must be wrong with libnm-gtk .la file?

**Comment by lethalman** [comment]
Also can't you just use networkmanagerapplet as build input for plugins etc. instead of that libnm-gtk? The plugins derivations are not needed in networkmanagerapplet.

**Comment by cstrahan** [comment]
@lethalman 

> Also can't you just use networkmanagerapplet as build input for plugins etc. instead of that libnm-gtk? The plugins derivations are not needed in networkmanagerapplet.

Those symlink lines caused a dependency cycle, because each of those plugins depend on libnm-gtk.so, which is also built as part of networkmanagerapplet.

> Those symlinks are done in nixos and I think should be just removed from this expression, not even commented out.

Ah, I thought the *.so were `dlopen`ed. I can just nuke those, then. Any strong feelings about keeping the two packages split up, or should we just keep it as one package?

File: pkgs/tools/networking/network-manager-applet/default.nix (hunk 1 above)
Comment (cstrahan) [inline pkgs/tools/networking/network-manager-applet/default.nix]: Nah, this was just so I could split-out libnm-gtk from the applet.

**Comment by cstrahan** [comment]
For my own reference: https://github.com/lethalman/nixpkgs/commit/b9a942ce9ac8cce78e5ec23e1ba647c5ced24578

**Comment by cstrahan** [comment]
@lethalman @iElectric

I *think* this should be good to go. I don't (yet) use NetworkManager, so I might ask either of you to give this a test.

(I wrote this PR because I'm thinking about trying NetworkManager instead of wpa_supplicant, but only if I can have `nmtui`.)

File: pkgs/tools/networking/network-manager/openconnect.nix [hunk 2]
```diff
@@ -1,18 +1,18 @@
 { stdenv, fetchurl, openconnect, intltool, pkgconfig, networkmanager
-, withGnome ? true, gnome3, procps, module_init_tools }:
+, withGnome ? true, libsecret-unstable, gnome3, procps, module_init_tools, libnm-gtk }:
```
Comment (lethalman) [inline pkgs/tools/networking/network-manager/openconnect.nix]: You probably forgot to use networkmanagerapplet here instead of libnm-gtk.

**Comment by lethalman** [comment]
Damn I think you were right, I was fooled by these symlinks: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/networkmanager.nix#L136

But they do not include the libs. So I guess your previous libnm-gtk was the right approach, sorry :-( I will test it once I go back home, unless @iElectric does it first.

**Comment by lethalman** [comment]
So I have tested the libnm-gtk version. The NM applet shows nothing, and journalctl tells me:
```
an 14 21:01:08 nixos dbus[1664]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Jan 14 21:01:08 nixos systemd[1]: [/nix/store/f4b1zkcrxg7sd8x8dmbykpv2bpnzcq2w-network-manager-1.0.0/etc/systemd/system/dbus-org.freedesktop.nm-dispatcher.service:1] Assignment outside of section. Ignoring.
Jan 14 21:01:08 nixos systemd[1]: dbus-org.freedesktop.nm-dispatcher.service lacks both ExecStart= and ExecStop= setting. Refusing.
Jan 14 21:01:08 nixos dbus[1664]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.nm-dispatcher.service': Unit dbus-org.freedesktop.nm-dispatcher.service failed to load: Invalid argument. See system logs and 'systemctl status dbus-org.freedesktop.nm-dispatcher.service' for details.
```

So something must be missing.

**Comment by lethalman** [comment]
Ok removing the `echo "Alias=..." >>` line in the networkmanager package seems to be the trick. I don't think that line is needed in any way.
However the problem remains, my gnome shell is unable to even show the wifi icon. However nmtui seems to work. Maybe it's a matter of upgrading gnome to 3.14, but I don't see any error so far.

**Comment by lethalman** [comment]
I don't see any applet with gnome 3.14 either :(

**Comment by lethalman** [comment]
I've tried this code in gjs:
```
NMClient.Client.new_async(null, function(obj, result) {
  var client = NMClient.Client.new_finish(result);
  var conn = client.get_primary_connection();
  var dev = conn.get_devices()[0];
  print(dev.interface);
});
```

And I get `wlp3s0` which is my primary wifi connection. Also from the client I get that the manager is running and that networking is active. That's roughly the code of gnome-shell applet, so I can't understand why it doesn't show the applet :(

**Pull request closed without merging**